### ISSUE TITLE
PHS CSS style

### ISSUE THREAD

**Issue by Moohan** [Public-Health-Scotland/phstemplates#66]
This possibly belongs in `phsstyles` rather than here.

It would be good to have a set of CSS styles that can be added to an RMarkdown doc (I think it already exists for the Shiny app). To apply standard PHS colours, formatting etc.

I combined the Shiny version with someone else's, and have been applying this one to docs.
<details>
  <summary>style.css</summary>

```CSS
/* Setting header colour */
  h1, .h1, h2, .h2, h3, .h3, h4, .h4, h5, .h5, h6, .h6 {
    margin-top: 20px;
    margin-bottom: 10px;
    color: #3F3685;
  }

/* Body styles */
  body {
    font-family: Arial;
    font-size: 16px;
    line-height: 1.5;
    color: #000000;
    background-color: #ffffff;
  }

/* Style hyperlinks */
  .externallink {color: #3F3685; font-weight: bold;}
.externallink:hover {color: #8FBFC2;}

.tocify {
    width: 20%;
    max-height: 90%;
    overflow: auto;
    margin-left: 2%;
    position: fixed;
    border: 4px solid #3F3685;
    border-radius: 15px;
}

.tocify .tocify-item a, .tocify .list-group-item {
    padding: 4px;
}

.list-group-item.active, .list-group-item.active:focus, .list-group-item.active:hover {
    z-index: 2;
    color: #fff;
    background-color: #655E9D;
    border-color: #004785;
    text-decoration: underline;
}
```
</details>

It would be good to have a 'template', so you can create a new RMarkdown doc which has this in the YAML header.

```yaml
output: 
  html_document:
    css: style.css
    toc: true
    toc_float: true
```

It might also be useful to have an Addin that would create the `style.css` file in the current directory, to be used as needed.

**Comment by alan-y** [comment]
@Moohan - this sounds fine, if you (any others with an interest) definitely think the best place for it is `phstemplates`, then I can look into developing this. Let me know when you are sure.

**Comment by Moohan** [comment]
My opinion is that it fits here as it would be an RMarkdown / CSS template. I only thought of phsstyles as that applies the PHS style to plots, in the same way, this idea would 'apply' a style to a document. 🤷‍♂️

**Comment by elliejwbatesphs** [comment]
I would certainly find this useful for some training materials I'm working on and other work I am doing at the moment.

Apologies, don't know css at all so could offer to test it is working if that helps, but not to develop.

**Comment by rubenv95** [comment]
Hi all, a few examples of CSS I have used for HTML docs if you want to consider incorporating? There are also plenty of other ideas that could be added for a shiny css template, but I don't think that fits into the same function. I think the shiny template that currently exists does very minimal changes to css, so definitely scope to improve this. 

I've attached a examples css script with the below, but it's lifted from a range of different projects over the last few years, so there isn't necessarily loads of consistency and a few don't use the root var(--[color]) approach, so that would need updated. Also being careful with how it affects other bits of css in an HTML doc - .a for example will update a lot of like in-built hyperlinks, such as some sidebars. 

[css examples txt file.txt](https://github.com/Public-Health-Scotland/phstemplates/files/12844686/css.examples.txt.file.txt)

Value Boxes (.small-box)
These are notoriously annoying to add to HTML documents, as I think the function does not include adding css styles. These need to be manually added, and I have some code that "hacks" the colours to make them PHS style. The drawback is there is a lot of CSS code to go with it.

Tabset panels (.nav-pills or .nav-tabs)
I have some css that updates a pills style tabset panel - regularly used in HTML docs.

Button styles (.btn)
Also easily implemented PHS purple buttons, these are the buttons that might open a modal, hyperlink, etc. More common on a shiny app, but they fit into HTML documents too. 

Hyperlinks (.a)
I think just adding PHS purple colour and removing text-decoration is nice, and matches PHS Templates for publications.
